sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py

```diff
@@ -244,7 +244,10 @@ def _build_pipeline_outputs(self, outputs: typing.Dict[str, NodeOutput]):
         for key, value in outputs.items():
             if not isinstance(key, str) or not isinstance(value, NodeOutput) or value._owner is None:
                 raise UserErrorException(message=error_msg, no_personal_data_message=error_msg)
-            meta = value._meta or value
+            if value._meta is not None:
+                meta = value._meta
+            else:
+                meta = value
```

Comment:
What for? I assumed meta always Input/Output class previously, indicate the definition of io, here seems meta is possibly a NodeOutput.

Reply:
yes, it seems so; and is there any other related places need to take care?

Reply:
Not sure, maybe have a discussion with you after I browsed all the files, just feels like put the NodeOutput into meta doesn't match the name 'meta', should be definition for 'meta'.